### DISCUSSION TITLE
websocket.close throws runtime error on a closed websocket.

### DISCUSSION THREAD

**Discussion by tobiashagge** [encode/starlette#2714] [Potential Issue] [answered]
I have the following situation:

1) I call 'await websocket.close(...)
2) The awaitable gets its turn on the async queue and calls 'await websocket.send(...)' to send a 'websocket-close' message.
3) Some time after step (1) but before step (4), the websocket gets closed.
4) The second awaitable gets its turn on the async queue and tries to send the message.
5) websocket.send throws a RuntimeError because the socket is closed.

This doesn't seem like correct behavior. Shouldn't it be either:
1) websocket.close catches the exception silently and returns without complaint, because there is nothing more to do, -or-
2) websocket.send throws a starlette-specific exception, either WebsocketDisconnect or something new.

**Comment by Kludex** [comment, marked as answer]
Bump starlette and uvicorn to the latest release. This behavior should be old.
